lineapy/api/api.py

```diff
@@ -70,10 +70,16 @@ def save(reference: object, name: str) -> LineaArtifact:
         returned value offers methods to access
         information we have stored about the artifact (value, version), and other automation capabilities, such as :func:`to_pipeline`.
     """
-    execution_context = get_context()
-    executor = execution_context.executor
-    db = executor.db
-    call_node = execution_context.node
+    logger.warn("Calling save inside control blocks is not supported")
+
+
+def _save(
```

Comment:
Use of these hidden functions makes sense if one looks at what `LineaTransformer` does (e.g., detection of `save()` and delegation to `_save()`), but it does not make much sense in this `api.py` script alone. Which makes me wonder if this will introduce a layer of confusion/complication for future contribution and maintenance efforts.

Is this refactoring a necessity? Or, can we refactor a bit differently such that the public API (e.g., `save()`) embeds/uses the corresponding hidden method (e.g., `_save()`) -- rather than having the two separated out?

Reply:
This was a happy coincidence. Refactoring these was part of the proposal so that we dont run them inside blackboxes but here, I needed to pass extra args to these commands which is why i had to change the signatures (get_context would not work because i'm not going thr the usual executor path. an alternate is to have a custom set_context that does the same thing but i feel that roundabout execution is not required plus i get to demo how we can stop these calls inside blackboxes so went this route) . This does not reflect how the final api.py will look though. most likely these _xxxx functions will be pulled into a separate internal_api.py or such file.

Reply:
Yeah, that makes good sense! I think we can discuss details of the refactoring after this prototyping phase.